packages/base-types/src/node/carousel.ts

```diff
@@ -27,6 +27,7 @@ export enum CarouselLayout {
 export interface CarouselButton extends DataID {
   name: string;
   intent?: Nullable<string>;
+  url?: Nullable<string>;
```

Comment:
I don't think we should do it this way. I guess we should just update the `NodeCarouselCard` type to look like this:

```typescript
export type NodeCarouselCard = CarouselCard<GeneralRequestButton | ActionRequestButton>;
```

this way we can reuse logic on prototype tool and on runtimes